src/sql/mir.rs

```diff
@@ -761,6 +761,28 @@ impl SqlToMirConverter {
                 }
             }
 
+
+            // 3. Get columns used by each filter node.
+            let mut filter_columns = HashMap::new();
+            let mut filter_nodes = Vec::new();
+            let mut moved_filters = Vec::new();
+            let mut sorted_rels: Vec<&String> = qg.relations.keys().collect();
+            sorted_rels.sort();
+            for rel in &sorted_rels {
+                let qgn = &qg.relations[*rel];
+                if *rel != "computed_columns" {
+                    if !qgn.predicates.is_empty() {
```

Comment:
I think it can be removed in this case (where we're only extracting relevant columns), although not in the case linked (where other datastructures are modified inside the the conditional but outside the loop).